GitOut/Features/Git/Diff/GitDiffResult.cs

```diff
@@ -66,9 +101,38 @@ public void Feed(string line)
                         header = string.Join("\r\n", parts);
                     }
                     parts.Clear();
+                    IsBinaryFile = true;
                     hasCreatedHeader = true;
                 }
                 parts.Add(line);
+                return this;
+            }
+
+            private static bool IsBinary(Stream stream)
+            {
+                const int limit = 8000;
+                byte[] buffer = new byte[limit];
+                int read;
+                int position = 0;
+                while ((read = stream.Read(buffer)) > 0)
+                {
+                    int maxCharacters = limit - position;
+                    for (int i = 0; i < maxCharacters && i < read; ++i)
+                    {
+                        if (buffer[i] == '\0')
```

Comment:
a file is considered binary if it contains a nullchar? 🤔 
googling this indicates that anything not ascii might be considered binary, i.e. `if (buffer[i] > 127) return true;`